lib/cretonne/meta/cdsl/ast.py

```diff
@@ -332,14 +368,51 @@ def inst_predicate(self):
         return pred
 
     def copy(self, m):
-        # type: (Dict[Var, Var]) -> Apply
+        # type: (VarMap) -> Apply
         """
         Return a copy of this Expr with vars replaced with fresh variables,
         in accordance with the map m. Update m as neccessary.
         """
         return Apply(self.inst, tuple(map(lambda e: replace_var(e, m),
                                           self.args)))
 
+    def vars(self):
+        # type: () -> Set[Var]
+        """ Return the set of all Vars that appear in self"""
```

Comment:
Should it be clarified that this is only the Vars corresponding to SSA values?

Vars can also refer to immediate operands:

```python
expand.legalize(
        a << icmp_imm(cc, x, y),
        Rtl(
            a1 << iconst(y),
            a << icmp(cc, x, a1)
        ))
```